src/lib/utils/SynapseClient.tsx

```diff
@@ -144,6 +148,44 @@ export const doGet = (
   return fetchWithExponentialTimeout(endpoint + url, options, 1000, 5)
 }
 
+export const doPut = (
+  url: string,
+  requestJsonObject: any,
+  sessionToken: string | undefined,
+  initCredentials: string | undefined,
+  endpoint: string
+) => {
+  const options: any = {
+    body: JSON.stringify(requestJsonObject),
+    headers: {
+      Accept: '*/*',
+      'Access-Control-Request-Headers': 'sessiontoken',
+      'Content-Type': 'application/json',
+    },
+    method: 'PUT',
+    mode: 'cors'
+  }
+  if (initCredentials) {
+    options.credentials = initCredentials
+  }
+  if (sessionToken) {
+    options.headers.sessionToken = sessionToken
+  }
+  return fetchWithExponentialTimeout(endpoint + url, options, 1000, 5)
+}
+
+export const putRefreshSessionToken = (sessionToken: string, endpoint: string = DEFAULT_ENDPOINT) => {
+  return doPut('/auth/v1/session', { sessionToken }, undefined, undefined, endpoint)
+}
+
+export const putSessionTokenAndTermsOfUse = (
```

Comment:
Please remove the ability to set terms of use from this utility.  A portal developer may want to use it, and they should not set this to true unless the user agrees to the full Synapse ToU (so the portal would need to properly present it).